src/Etg.Yams.Core/Storage/Config/DeploymentConfig.cs

```diff
@@ -1,305 +1,184 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 using System.Linq;
 using Etg.Yams.Application;
-using Etg.Yams.Utils;
 
 namespace Etg.Yams.Storage.Config
 {
-    public class DeploymentConfig
+    public class DeploymentConfig : IEnumerable<AppDeploymentConfig>
     {
-        private readonly IReadOnlyDictionary<string, AppDeploymentConfig> _apps =
-            new Dictionary<string, AppDeploymentConfig>();
+        private readonly ISet<AppDeploymentConfig> _apps = new HashSet<AppDeploymentConfig>();
 
         public DeploymentConfig()
         {
         }
 
-        private DeploymentConfig(IDictionary<string, AppDeploymentConfig> apps)
+        protected bool Equals(DeploymentConfig other)
         {
-            _apps = new ReadOnlyDictionary<string, AppDeploymentConfig>(apps);
+            return Equals(_apps, other._apps);
         }
 
-        public DeploymentConfig(string rawData)
+        public override bool Equals(object obj)
         {
-            _apps = ParseDeploymentsConfig(rawData);
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((DeploymentConfig) obj);
         }
 
-        public IEnumerable<string> ListApplications()
+        public override int GetHashCode()
         {
-            return _apps.Keys;
+            return (_apps != null ? _apps.GetHashCode() : 0);
```

Comment:
I'll test this a bit more. I generated it the implementation with Resharper. I'd expect that two sets with the same values would have the same hashcode (GetHashCode is supposed to be consistent with Equals) but I'll make sure.